### ISSUE TITLE
`gl_shader_util` "Assertion Failed" error when launching `citra.exe` from the terminal.

### ISSUE THREAD

**Issue by TheV360** [citra-emu/citra#6009]
I'm using Citra 4a9995a (1761, 2022-04-23), and I'm launching `citra.exe` directly with the following command:
```bash
/the/path/to/citra.exe /another/path/to/AnyOldCXIDump.cxi
```

Here's the log:
[citra_log_but_for_cxi.txt](https://github.com/citra-emu/citra/files/8548723/citra_log.txt)

And the log doesn't even contain the actual error, so here's the last bit of the terminal output:
```
[   1.508725] Render.OpenGL <Info> video_core/renderer_opengl/renderer_opengl.cpp:Init:1261: GL_VERSION: 3.3.0 NVIDIA 472.84
[   1.508731] Render.OpenGL <Info> video_core/renderer_opengl/renderer_opengl.cpp:Init:1262: GL_VENDOR: NVIDIA Corporation
[   1.508732] Render.OpenGL <Info> video_core/renderer_opengl/renderer_opengl.cpp:Init:1263: GL_RENDERER: NVIDIA GeForce RTX 2070 with Max-Q Design/PCIe/SSE2
[   1.510031] Render.OpenGL <Info> video_core/renderer_opengl/gl_format_reinterpreter.cpp:FormatReinterpreterOpenGL:378: Using shader for D24S8 to RGBA8 reinterpretation
[   1.514698] Loader <Info> core/loader/ncch.cpp:Load:192: Program ID: 0004000000175E00
[   1.541020] Loader <Info> core/hle/kernel/process.cpp:ParseKernelCaps:141: ExHeader kernel version: 2.50
[   1.543582] Service.CFG <Info> core/hle/service/cfg/cfg.cpp:SetPreferredRegionCodes:656: Preferred region code set to 1
[   1.550219] Render.OpenGL <Info> video_core/renderer_opengl/gl_shader_disk_cache.cpp:LoadTransferable:169: Found a transferable disk cache with 49 entries
[   1.550308] Render.OpenGL <Info> video_core/renderer_opengl/gl_shader_disk_cache.cpp:LoadPrecompiled:180: No precompiled shader cache found for game with title id=0004000000175E00
[   1.791173] Debug <Critical> video_core/renderer_opengl/gl_shader_util.cpp:operator():106: Assertion Failed!
Shader not linked
```
For convenience, here's a direct link to that assertion:
https://github.com/citra-emu/citra/blob/1382035d4de5ada310c7f4fd00f1836b16ab74fb/src/video_core/renderer_opengl/gl_shader_util.cpp#L106

This command that I'm using used to work, too -- I think it last worked around February of this year.

I tested this with my dumps of Pokemon Moon and SmileBASIC, and both of them behave in the same way.

I tested this with `*.3ds` dumps, and those get another error and die in a different way:
[citra_log_but_for_3ds.txt](https://github.com/citra-emu/citra/files/8548732/citra_log.txt)
Maybe that's out of the scope for this issue, but :shrug:.

I did see that it was mentioning something about shaders, so I forced `citra.exe` to run on my Nvidia GPU rather than my iGPU. That didn't do anything.

Either way, this **isn't too urgent**, as I can just launch the dumps this way:
```bash
/path/to/citra-qt.exe /path/to/AnyOldCXIDump.cxi
```

**Comment by SachinVin** [comment]
I think I know the issue, probably introduced in https://github.com/citra-emu/citra/pull/5976

**Comment by SachinVin** [comment]
Fixed with #6083
